### PR TITLE
Enable behat tests for Channel after upgrade to APIP 3

### PR DESCRIPTION
**Pull request by TheMilek**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | api-platform-3                   |
| Bug fix?        | no                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| License         | MIT                                                          |

continuation of https://github.com/Sylius/Sylius/pull/16126

### CODE REVIEW

File: features/admin/channel/managing_channels/adding_channel_with_set_period_of_showing_lowest_price.feature [hunk 1]
```diff
@@ -10,7 +10,7 @@ Feature: Specifying the lowest price for discounted products checking period whi
         And the store operates in "United States" and "Poland"
         And I am logged in as an administrator
 
-    @api @ui
+    @todo-api @ui
```
Comment (TheMilek) [inline features/admin/channel/managing_channels/adding_channel_with_set_period_of_showing_lowest_price.feature]: To be enabled with ChannelPriceHistoryConfig resource upgrade

File: features/admin/channel/managing_channels/selecting_default_tax_zone_for_channel.feature [hunk 2]
```diff
@@ -8,7 +8,7 @@ Feature: Selecting default tax zone for a channel
         Given the store operates on a single channel in "United States"
         And I am logged in as an administrator
 
-    @api @ui
+    @todo-api @ui
```
Comment (TheMilek) [inline features/admin/channel/managing_channels/selecting_default_tax_zone_for_channel.feature]: To be enabled with Zone resource upgrade

File: features/admin/channel/managing_channels/editing_menu_taxon_on_channel.feature [hunk 3]
```diff
@@ -11,7 +11,7 @@ Feature: Editing menu taxon on channel
         And channel "Web Store" has menu taxon "Clothes"
         And I am logged in as an administrator
 
-    @api @ui @javascript
+    @todo-api @ui @javascript
```
Comment (TheMilek) [inline features/admin/channel/managing_channels/editing_menu_taxon_on_channel.feature]: To be enabled with taxon resource upgrade

File: features/admin/channel/managing_channels/adding_new_channel_with_menu_taxon.feature [hunk 4]
```diff
@@ -11,7 +11,7 @@ Feature: Adding a new channel with menu taxon
         And the store classifies its products as "Clothes" and "Guns"
         And I am logged in as an administrator
 
-    @ui @javascript @api
+    @todo-api @ui @javascript
```
Comment (TheMilek) [inline features/admin/channel/managing_channels/adding_new_channel_with_menu_taxon.feature]: Needs taxon resource upgrade

File: features/shop/channel/filtering_out_disabled_channels.feature [hunk 5]
```diff
@@ -11,7 +11,7 @@ Feature: Filtering out disabled channels
         And there is product "Old Wardrobe" available in "Furniture" channel
         And the channel "Fashion" is disabled
 
-    @ui @api
+    @ui @todo-api
```
Comment (TheMilek) [inline features/shop/channel/filtering_out_disabled_channels.feature]: Needs product resource upgrade

File: src/Sylius/Behat/Context/Api/Admin/ManagingChannelsContext.php [hunk 6]
```diff
@@ -111,10 +109,9 @@ public function iSpecifyItsAs(string $field, string $value = ''): void
      */
     public function iChooseAsTheBaseCurrency(?CurrencyInterface $currency = null): void
     {
-        $this->client->addRequestData(
-            'baseCurrency',
-            null === $currency ? $currency : $this->sectionAwareIriConverter->getIriFromResourceInSection($currency, 'admin'),
-        );
+        if ($currency !== null) {
+            $this->client->addRequestData('baseCurrency', $this->iriConverter->getIriFromResource($currency));
+        }
```
Comment (NoResponseMate) [inline src/Sylius/Behat/Context/Api/Admin/ManagingChannelsContext.php:114]: Passing a `null` fails on IRI conversion. We should investigate whether `PATCH` is still the default way of updating resources in APIP3.

**Review by NoResponseMate — APPROVED** [review]

**Comment by NoResponseMate** [comment]
Thank you, @TheMilek!

**Pull request merged**